libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,662 @@
+    */
+    navigationIndicator: boolean;
+
+    /**
+     * Whether By line is present in list item*/
+    hasByLine: boolean;
+
+    /**@hidden
+     * To display loading symbol */
+    loading = false;
+
+    /**
+     * @hidden
+     * keyManger to handle keybord events
+     * used in template
+     */
+    keyManager: FocusKeyManager<BaseListItem>;
+
+    /** The model backing of the component. */
+    selectionModel: SelectionModel<BaseListItem>;
+
+    /**@hidden
+     * Whether list component has multiselection
+     * binding in tempate to append class */
+    multiSelect = false;
+
+    items = [];
+
+    selectedvalue: string;
+
+    /** @hidden */
+    _contentDensity = this._listConfig.contentDensity;
+
+    /**
+     * @hidden
+     * Used to define if contentDensity value is 'compact' or not.
+     */
+    isCompact = this._contentDensity === 'compact';
+
+    /** @hidden
+    * To store */
+    _tempItems = [];
+
+    /** @hidden
+    */
+    _startIndex = 0;
+
+    /** @hidden
+    */
+    _lastIndex = this.itemSize;
+
+    /** @hidden
+    */
+    _dsItems = [];
+
+    /** @hidden
+    */
+    _destroyed = new Subject<void>();
+
+    /**@hidden
+     * for data source handling
+     */
+    _dsSubscription: Subscription | null;
+
+    /**@hidden
+    * for items impertaive and declartive approaches
+    */
+    _itemsSubscription: Subscription | null;
+
+    /**
+    * Child items of the List.
+    */
+    @ContentChildren(BaseListItem, { descendants: true })
+    ListItems: QueryList<BaseListItem>;
+
+    @ContentChild(ListItemDef)
+    listItemDef: ListItemDef;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    /** role */
+    @HostBinding('attr.role')
+    role = 'list';
+
+    /** @hidden */
+    @Output()
+    selectedItemChange: EventEmitter<SelectionChangeEvent> = new EventEmitter<SelectionChangeEvent>();
+
+    /**
+    * Datasource for suggestion list
+    */
+    @Input()
+    get dataSource(): FdpListDataSource<any> {
+        return this._dataSource;
+    }
+    set dataSource(value: FdpListDataSource<any>) {
+        if (value) {
+            this._initializeDS(value);
+        }
+    }
+    protected _dataSource: FdpListDataSource<any>;
+
+    /**
+    * content Density of element. 'cozy' | 'compact'
+    */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
+    }
+
+    /** setter and getter for _navigated */
+    @Input('navigated')
+    get navigatedValue(): boolean {
+        return this.navigated;
+    }
+
+    set navigatedValue(value: boolean) {
+        this.navigated = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation');
+    }
+
+    /** setter and getter for _navigationIndicator */
+    @Input('navigationIndicator')
```

Comment:
The setter and getter should be named differently. 

its it should be always navigationValue, you should not have different names for the same thing


where you create private _navigationIndicator so names done collide. 

If you need to have this public the `_navigationIndicator` then have this as public after all decorators (and mark it as hidden)

There is Issue opened to update coding guidelines, where we have some private API property that needs to be accessed from template, then it needs to be public but marked with `_` prefix to tell this is private API.

Reply:
done